src/main/java/io/confluent/kafkarest/resources/TopicsResource.java

```diff
@@ -74,6 +58,37 @@ public Topic getTopic(@PathParam("topic") String topicName) {
     return topic;
   }
 
+  @GET
+  @Path("/{topic}/partition/{partition}/messages")
+  @PerformanceMetric("topic.consume-binary")
+  @Produces({Versions.KAFKA_V1_JSON_BINARY_WEIGHTED,
+             Versions.KAFKA_V1_JSON_WEIGHTED,
+             Versions.KAFKA_DEFAULT_JSON_WEIGHTED,
+             Versions.JSON_WEIGHTED,
+             Versions.KAFKA_V1_JSON_BINARY_WEIGHTED_LOW,
```

Comment:
The `WEIGHTED_LOW` versions shouldn't go here, the should go on the `@Produces` annotation on the class. They represent the same content type as other entries but at a lower preference level used during content negotiation. So `KAFKA_V1_JSON_BINARY_WEIGHTED` is effectively `application/vnd.kafka.binary.v1+json; qs=1.0` whereas `KAFKA_V1_JSON_BINARY_WEIGHTED_LOW` is `application/vnd.kafka.binary.v1+json; qs=0.1`. We put the `WEIGHTED_LOW` version on the class because people are bound to just set one Accept header for all their requests, especially on a single resource. Adding it at low weight tries to discourage it (the server will only use it if it has no other choice), but still permits it.